IPLD-Data-Model-v1.md

```diff
@@ -1,17 +1,103 @@
-# [WIP] IPLD Data Model
+IPLD Data Model
+===============
 
-## Simple Types
+The IPLD Data Model is "layer 1" of the IPLD core models.
+("Layer 0" are the [Codecs](Codecs); "layer 2" is called [Schemas](schema).)
+
+The Data Model specifies what categories of data are representable in IPLD --
+for example, booleans, and integers, and textual strings.
+[Codecs](Codecs) specify exactly how these data are transcribed into serialized bytes.
+
+
+Kinds
+-----
+
+The following is the list of essential _kinds_ (or more formally, _representation kinds_)
+of data representable in the IPLD Data Model:
 
-* Boolean
 * Null
+* Boolean
+* Integer
+* Float
 * String
+* Bytes
+* List
+* Map
+* Link
+
+(Note that we use the term "_kinds_" here to disambiguate this from "_types_",
```

Comment:
For some reason I instinctively consider "type" lower level than "kind", maybe that's just me tho. I totally agree we need a way to distinguish between these two things, I think I've said "primitive types" and "base types" before and those terms don't distinguish between them enough.

Reply:
I agree, _but_, if the schema work goes well then we're going to be talking about their "types" more than these representation "kinds", so which term would you rather be using more often?

Reply:
That’s a good point. Type is the more well known term so I’d prefer to use that one more often.